File: Baekjoon/11727.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int length = Integer.parseInt(br.readLine());
        
        int d[] = new int[length + 1];
        d[0] = 1;
        d[1] = 1;
        for (int i = 2; i <= length; i++){
            d[i] = d[i - 1] + 2 * d[i - 2];
            d[i] %= 10007;       
        }
        System.out.println(d[length]);
    }
}

File: ETC/11727.kt
import java.io.*

fun main() {
    val br = System.`in`.bufferedReader()
    
    val n = br.readLine().toInt()
    
    val dp = IntArray(1001)
    
    dp[1] = 1
    dp[2] = 3
    
    for (i in 3..n) {
        dp[i] = (dp[i - 1] + 2 * dp[i - 2]) % 10007
    }
    
    println(dp[n])
}
